### ISSUE TITLE
Glitch with the trend - 7 increasing/ decreasing points 

### ISSUE THREAD

**Issue by hp0709** [nhs-r-community/NHSRplotthedots#171]
The rule for trend, 7 points increasing /decreasing labels the colours based on the side of the mean rather than the direction. Also highlights other points not part of the trend?

[Test_Data_rule3_Exmaple2.xlsx](https://github.com/nhs-r-community/NHSRplotthedots/files/10392652/Test_Data_rule3_Exmaple2.xlsx)
[Test_Data_rule3_Exmaple1.xlsx](https://github.com/nhs-r-community/NHSRplotthedots/files/10392654/Test_Data_rule3_Exmaple1.xlsx)
i've added two sample datasets where you can see this issue. 

```
p <- ptd_spc(.data = df, value_field = "Pre5",date_field = "ReportingDate", improvement_direction = "decrease") 

p
```

**Comment by tomjemmett** [comment]
reviewing data in example1, our code generates the following:
![image](https://github.com/nhs-r-community/NHSRplotthedots/assets/12023696/b3ee6f00-9c6f-4e75-b4db-54f485b23075)

Compared to the NHSE Excel Tool:

![image](https://github.com/nhs-r-community/NHSRplotthedots/assets/12023696/58eae40c-1229-4493-b863-76ec7c80145d)

even more interesting is if we change to increase improvement direction:
![image](https://github.com/nhs-r-community/NHSRplotthedots/assets/12023696/06cb2f73-75ee-42ed-8958-5044d5b7bc1b)

vs

![image](https://github.com/nhs-r-community/NHSRplotthedots/assets/12023696/799cd91e-0761-4902-8da3-0c8fdc1866e7)


Looking at the code in `ptd_calculate_point_type` we take a simple approach of working out whether the point is above the mean or below the mean, so trends of 7 are flagged as concern incorrectly. This needs to be rewritten, which may also solve #122

**Comment by tomjemmett** [comment]
I've made out code match the last screenshot from above, which is the "increasing"/"high is good" version. For the "decreasing"/"low is good" version we end up with a direct inversion of the colouring for the "increasing"/"high is good" chart.

This is in contrast to the Excel tool which (as can be seen above) does something somewhat unpredictable for the "low is good" case with the sample data provided in this issue. Attached is a copy of the Excel tool with the sample data. I think that the logic implemented in #180 is at least clear and well-documented in case we have any questions of deviations in future.

[NHSE_MDC_XmR.zip](https://github.com/nhs-r-community/NHSRplotthedots/files/12064259/NHSE_MDC_XmR.zip)
